numpyro/distributions/transforms.py

```diff
@@ -147,7 +155,21 @@ def codomain(self):
 
     @property
     def event_dim(self):
-        return max(p.event_dim for p in self.parts)
+        raise ValueError("Please use `.input_event_dim` or `.output_event_dim` instead.")
+
+    @property
+    def input_event_dim(self):
+        input_event_dim = self.parts[-1].input_event_dim
+        for part in self.parts[len(self.parts) - 1::-1]:
+            input_event_dim = part.input_event_dim + max(input_event_dim - part.output_event_dim, 0)
```

Comment:
Will this ever be anything other than `part[0].input_event_dim` in practice?

Reply:
Yes, as in the tests, we have `transforms.AffineTransform(0, 2).event_dim = 0` but the composed one will have `input_event_dim = 1`. In general, `Transform.inv` should revert `input_event_dim` and `output_event_dim`, but it is clear that after composed, `output_event_dim != parts[-1].output_event_dim`. This preserves the symmetric.

Reply:
I see, that makes sense!